tcms/issuetracker/types.py

```diff
@@ -384,6 +386,96 @@ def report_issue_from_testcase(self, caserun):
         return url + '/issues/new?' + urlencode(args, True)
 
 
+class Redmine(IssueTrackerType):
+    """
+        Support for Redmine. Requires:
+
+        :api_url: - the API URL for your Redmine instance
+        :api_username: - a username registered in Redmine
+        :api_password: - the password for this username
+    """
+
+    def __init__(self, tracker):
+        super().__init__(tracker)
+
+        if not self.is_adding_testcase_to_issue_disabled():
+            self.rpc = redminelib.Redmine(
+                self.tracker.api_url,
+                username=self.tracker.api_username,
+                password=self.tracker.api_password
+            )
+
+    def add_testcase_to_issue(self, testcases, issue):
+        for case in testcases:
+            redmine_integration.RedmineThread(self.rpc, case, issue).start()
+
+    def all_issues_link(self, ids):
+        if not self.tracker.base_url:
+            return None
+
+        if not self.tracker.base_url.endswith('/'):
+            self.tracker.base_url += '/'
+
+        query = 'issues?utf8=✓&set_filter=1&sort=id%3Adesc&f%5B%5D=issue_id'
+        query += '&op%5Bissue_id%5D=%3D&v%5Bissue_id%5D%5B%5D={0}'.format('%2C'.join(ids))
+        query += '&f%5B%5D=&c%5B%5D=tracker&c%5B%5D=status&c%5B%5D=priority'
+        query += '&c%5B%5D=subject&c%5B%5D=assigned_to&c%5B%5D=updated_on&group_by=&t%5B%5D='
+
+        return self.tracker.base_url + query
+
+    def find_project_by_name(self, caserun):
+        projects = self.rpc.project.all()
+        for prj in projects:
+            if prj.name == caserun.run.plan.product.name:
+                return prj
+
+        return None
+
+    @staticmethod
+    def find_issue_type_by_name(project, name):
+        for trk in project.trackers:
+            if str(trk).lower() == name.lower():
+                return trk
+
+        return None
+
+    def report_issue_from_testcase(self, caserun):
+        # note: Redmine instance needs to have the same projects
+        # defined otherwise this will fail!
+        project = self.find_project_by_name(caserun)
+
+        if not project:
+            return self.tracker.base_url
+
+        issue_type = self.find_issue_type_by_name(project, 'Bug')
+
+        if not issue_type:
+            issue_type = project.trackers[0]
+
+        query = "issue[tracker_id]=" + str(issue_type.id)
+        query += "&issue[subject]="
+        query += "{0}".format(quote('Failed test:{0}'.format(caserun.case.summary)))
```

Comment:
Drop the enclosing `format` invocation. Why not just `+= quote()` ?

Reply:
OK